mlflow/models/container/__init__.py

```diff
@@ -92,22 +93,53 @@ def _install_pyfunc_deps(model_path=None, install_mlflow=False):
             return
         conf = model.flavors[pyfunc.FLAVOR_NAME]
         if pyfunc.ENV in conf:
-            print("creating and activating custom environment")
-            env = conf[pyfunc.ENV]
-            env_path_dst = os.path.join("/opt/mlflow/", env)
-            env_path_dst_dir = os.path.dirname(env_path_dst)
-            if not os.path.exists(env_path_dst_dir):
-                os.makedirs(env_path_dst_dir)
-            shutil.copyfile(os.path.join(MODEL_PATH, env), env_path_dst)
-            conda_create_model_env = "conda env create -n custom_env -f {}".format(env_path_dst)
-            if Popen(["bash", "-c", conda_create_model_env]).wait() != 0:
-                raise Exception("Failed to create model environment.")
-            has_env = True
+            if no_conda:
+                print("Installing required dependencies")
+                env = conf[pyfunc.ENV]
+                env_path_dst = os.path.join("/opt/mlflow/", env)
+                env_path_dst_dir = os.path.dirname(env_path_dst)
+                if not os.path.exists(env_path_dst_dir):
+                    os.makedirs(env_path_dst_dir)
+                shutil.copyfile(os.path.join(MODEL_PATH, env), env_path_dst)
+                import yaml
+
+                # Generate requirements.txt file with the dependencies from the conda.yaml file.
+                with open(env_path_dst) as env_file:
+                    env_dep = yaml.full_load(env_file)
+                    dependency = env_dep["dependencies"]
+                    exclude = ["python", "mlflow"]
+                    with open('/opt/mlflow/requirements.txt', 'w') as req:
+                        for lib in dependency:
+                            if isinstance(lib, dict):
+                                for i in lib["pip"]:
+                                    if i not in exclude:
+                                        req.write(i+'\n')
+                            else:
+                                if lib.split('=')[0] not in exclude:
+                                    req.write(lib.replace("=", "==")+'\n')
+
+                    req.close()
+                env_file.close()
```

Comment:
Seems like this whole chunk should be extracted to a method.

```python
def _write_requirements_from_env(base_path="/opt/mlflow"):
    with open(os.path.join(base_path, conf[pyfunc.ENV])) as env_file:
        env_yaml = yaml.full_load(env_file)
        dependencies = env_yaml["dependencies"]
        exclude = ["python", "mlflow"]
        with open(os.path.join(base_path, "requirements.txt"), "w") as req_file:
            for dependency in dependencies:
                if isinstance(dependency, dict):
                    req_file.write(
                        "\n".join(d for d in dependency["pip"] if d not in exclude)
                        + "\n"
                    )
                else:
                    if dependency.split("=")[0] not in exclude:
                        req_file.write(dependency.replace("=", "==") + "\n")
```

Also, no need to manually `close` the files when using `with`.